external-crates/move/crates/move-vm-runtime/src/interpreter.rs

```diff
@@ -103,7 +108,7 @@ impl Interpreter {
     /// Entrypoint into the interpreter. All external calls need to be routed through this
     /// function.
     pub(crate) fn entrypoint(
-        function: Arc<Function>,
+        fun_ref: *const Function,
```

Comment:
would it make sense to use a ref `&` instead of the `*const` in signatures?
I am not sure it would introduce a lifetime annoyance but I think it would look better?
In many ways the `*const` is an implementation detail?

Reply:
Changing this to `ArenaPointer<Function>`, as clarified below.

Reply:
See the comment below about the follow-up PR